src/rgw/rgw_rest_s3.cc

```diff
@@ -1187,6 +1289,21 @@ int RGWPutObj_ObjStore_S3::get_params()
 
   } /* copy_source */
 
+  /* handle object tagging */
+  auto tag_str = s->info.env->get("HTTP_X_AMZ_TAGGING");
+  if (tag_str){
+    obj_tags.reset(new RGWObjTags());
+    ret = obj_tags->set_from_string(tag_str);
```

Comment:
We have a ceph::make_unique now, if you want it.

(But you don't have to use it.)